pkg/manifests/manifests.go

```diff
@@ -4165,19 +4167,26 @@ func (f *Factory) HashSecret(secret *v1.Secret, data ...string) (*v1.Secret, err
 func addRemoteWriteConfigs(clusterID string, rw []monv1.RemoteWriteSpec, rwTargets ...RemoteWriteSpec) []monv1.RemoteWriteSpec {
 	clusterIDRelabelConfig := []monv1.RelabelConfig{
 		{
-			TargetLabel: "__tmp_openshift_cluster_id__",
+			TargetLabel: tmpClusterIDLabelName,
 			Replacement: clusterID,
 		},
 	}
+	tmpRelabelDrop := monv1.RelabelConfig{
+		Regex:  tmpClusterIDLabelName,
+		Action: "labeldrop",
+	}
 
 	for _, target := range rwTargets {
+		// prepend our temporary cluster id label
+		writeRelabelConfigs := append(clusterIDRelabelConfig, target.WriteRelabelConfigs...)
```

Comment:
(nit) it might be more readable to group 

```suggestion
		// prepend our temporary cluster id label and drop it eventually
		writeRelabelConfigs := append(clusterIDRelabelConfig, target.WriteRelabelConfigs...)
		writeRelabelConfigs := append(writeRelabelConfigs, tmpRelabelDrop)
```